metashare/api/models.py

```diff
@@ -132,12 +133,45 @@ def __str__(self):
         return self.url
 
 
+class ProjectSlug(AbstractSlug):
+    parent = models.ForeignKey(
+        "Project", on_delete=models.PROTECT, related_name="slugs"
+    )
+
+
+class Project(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Model):
+    name = models.CharField(max_length=50)
+    description = MarkdownField(blank=True, property_suffix="_markdown")
+    branch_name = models.SlugField(max_length=50)
+
+    product = models.ForeignKey(
+        Product, on_delete=models.PROTECT, related_name="projects"
+    )
+
+    slug_class = ProjectSlug
+
+    def __str__(self):
+        return self.name
+
+    def save(self, *args, **kwargs):
+        if not self.branch_name:
+            self.branch_name = slugify(self.name)
+        super().save(*args, **kwargs)
+
+    class Meta:
+        ordering = ("-created_at", "name")
+        unique_together = (("name", "product"), ("branch_name", "product"))
```

Comment:
@wlonk If I try to create a new project with the same name but different capitalization as an existing project (on that product/repo), I get a `500` error:

```
IntegrityError at /api/projects/
duplicate key value violates unique constraint "api_project_branch_name_product_id_dd9a8e81_uniq"
DETAIL:  Key (branch_name, product_id)=(sample-project-54, 28) already exists.
```